operators/channel_offset.py

```diff
@@ -85,29 +78,119 @@ def poll(cls, context):
         return context.selected_sequences
 
     def execute(self, context):
-        selection = [s for s in context.selected_sequences if not s.lock]
+        
+        max_channel = 32
+        min_channel = 1
+
+        selection = [s for s in context.selected_sequences if (
+            not s.lock and not (self.direction == "down" and s.channel == min_channel)
+            and not (self.direction == "up" and s.channel == max_channel)
```

Comment:
Please avoid really long list comprehensions: they're hard to read. I would create two mini-functions: is_in_max_channel(sequence) and is_in_min_channel(sequence), something like that.